tests/cypress/integration/actions_objects2/case_16_z_order_features.js

```diff
@@ -70,7 +70,8 @@ context('Actions on polygon', () => {
         });
 
         it('Activate first shape', () => {
-            cy.get('#cvat_canvas_shape_1').trigger('mousemove').trigger('mouseover');
+            cy.get('#cvat-objects-sidebar-state-item-1').trigger('mousemove').trigger('mouseover');
```

Comment:
I had the same issue in another test when was writing PR #3417 (test case_37_object_make_copy.js). It also was reproducible only in headless. I changed the following:
![Screenshot from 2021-07-22 00-35-23](https://user-images.githubusercontent.com/40690378/126563256-477fad30-8a31-4f82-a8dc-dcf4a116084f.png)

And looks like it helped. 
Maybe it helps you to understand the root cause (I didn't understand).